internal/rbd/rbd_util.go

```diff
@@ -1333,6 +1334,31 @@ func lookupRBDImageMetadataStash(path string) (rbdImageMetadataStash, error) {
 	return imgMeta, nil
 }
 
+// updateRBDImageMetadataStash reads and updates stashFile with the required
+// fields at the passed in path, in JSON format.
+func updateRBDImageMetadataStash(path, device string) error {
+	if device == "" {
+		return fmt.Errorf("device is empty")
+	}
+	imgMeta, err := lookupRBDImageMetadataStash(path)
+	if err != nil {
+		return fmt.Errorf("failed to find image metadata: %w", err)
+	}
+	imgMeta.DevicePath = device
+
+	encodedBytes, err := json.Marshal(imgMeta)
+	if err != nil {
+		return fmt.Errorf("failed to marshall JSON image metadata for image (%s): %w", imgMeta.ImageName, err)
```

Comment:
`marshall` to `marshal` we need to return pool and radosnamespace name along with image name